src/Parsing.h

```diff
@@ -28,7 +28,7 @@ int64_t GetInteger64(void);
 /**
  * Parses an integer value.
  * Uses a method that obeys hexadecimal 0xN syntax needed for certain bitmasks.
- * self_destruct() will be called to abort when invalid tokens are encountered.
+ * A Cfg::FatalError exception will be thrown when invalid tokens are encountered.
```

Comment:
Are you sure? It seems to me that knowing that a specific exception may be thrown is relevant information for any caller of that code, and it doesn't contradict using defensive programming

Reply:
> Are you sure?

Yes, I am sure.

> It seems to me that knowing that a specific exception may be thrown is relevant information for any caller of that code, and it doesn't contradict using defensive programming

Yes, it is a common misconception. In reality, the vast majority of callers do not and should not care about specific exceptions. Proof by contradiction: If knowing possible exceptions is a good idea, then a simple function that calls 3 other functions which call 3 functions each may have to keep in mind 9 exceptions and somehow miraculously track when the list of those exceptions changes. Imagine what "relevant information" would become for a high-level function that calls (indirectly) 1000 functions... And for what? For nothing: Most functions can (and should) do absolutely nothing about specific exceptions. What may seem like "relevant information" at the first glance is actually distracting noise.

This noise is one of the reasons why C++ standard [changed](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2015/p0003r0.html#2.0) its approach to `throw` specification in function declarations:

> Dynamic exception specifications are a failed experiment, but this is not immediately clear to novices, especially where the "novice" is an experienced developer coming from other languages such as Java, where exception specifications may be more widely used. Their continuing presence, especially in an important part of the languge model (exceptions are key to understanding constructors, destructors, and RAII) is an impediment that must be explained. It remains embarassing to explain to new developers that there are features of the language that conforming compilers are required to support, yet should never be used. 

Informal documentation we are discussing here is just a reflection of the same "failed experiment" with the formal `throw` declaration. In some sense, documentation is even worse because no compiler or linting tool would warn the developer when that documentation becomes stale and lies.

Reply:
FTR, this was attempting to document how this function differed from the rest of the API (which use assert's), and document the parse-specific exception type it uses. Like C++ language documents context-specific exception outputs.

Done in [e4f9bf0](https://github.com/squid-cache/squid/pull/574/commits/e4f9bf074d0c560131168fb6f1705828996afbca).